File: src/Model/Devis.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Model;

/**
 *
 * @author helam
 */
public class Devis {
    ///var
    private int id_devis;
    GarageC garage=new GarageC();
    private int TVA=19;
    private float total;
    private User user;
    Maintenance maintenance=new Maintenance();
    
    
    
    ///constructeur
    public Devis() {
    }

    public Devis(User user, float total, int id_user) {
        this.id_devis = id_devis;
        this.total = total;
        this.user = user;
    }

    public Devis(float total, User user) {
        this.total = total;
        this.user = user;
    }


 
    
    ///toString
    @Override
    public String toString() {
        return "Devis{" + ", garage=" + garage + ", TVA=" + TVA + ", total=" + total + ", id_user=" + user+ ", maintenance=" + maintenance + '}';
    }

    /////getters & setters
    public int getId_devis() {
        return id_devis;
    }

    public void setId_devis(int id_devis) {
        this.id_devis = id_devis;
    }

    public GarageC getGarage() {
        return garage;
    }

    public void setGarage(GarageC garage) {
        this.garage = garage;
    }

    public int getTVA() {
        return TVA;
    }

    public void setTVA(int TVA) {
        this.TVA = TVA;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Maintenance getMaintenance() {
        return maintenance;
    }

    public void setMaintenance(Maintenance maintenance) {
        this.maintenance = maintenance;
    }

    public float getTotal() {
        return total;
    }

    public void setTotal(float total) {
        this.total = total;
    }


    
}

File: src/Entity/Devis.php
<?php

namespace App\Entity;

use App\Repository\DevisRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: DevisRepository::class)]
class Devis
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:"id_devis")]
    private ?int $id = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false,name:"id_user",referencedColumnName:"id_user")]
    private ?User $user = null;

    #[ORM\Column]
    private ?int $TVA = null;

    #[ORM\Column]
    private ?float $total = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false,name:"id_garage",referencedColumnName:"id_garage")]
    private ?Garagec $garage = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false,name:"id_maintenance",referencedColumnName:"id_maintenance")]
    private ?Maintenance $maintenance = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getTVA(): ?int
    {
        return $this->TVA;
    }

    public function setTVA(int $TVA): self
    {
        $this->TVA = $TVA;

        return $this;
    }

    public function getTotal(): ?float
    {
        return $this->total;
    }

    public function setTotal(float $total): self
    {
        $this->total = $total;

        return $this;
    }

    public function getGarage(): ?Garagec
    {
        return $this->garage;
    }

    public function setGarage(?Garagec $garage): self
    {
        $this->garage = $garage;

        return $this;
    }

    public function getMaintenance(): ?Maintenance
    {
        return $this->maintenance;
    }

    public function setMaintenance(?Maintenance $maintenance): self
    {
        $this->maintenance = $maintenance;

        return $this;
    }
}
